DICOMPlugins/DICOMM3DPlugin.py

```diff
@@ -0,0 +1,174 @@
+from SlicerDevelopmentToolboxUtils.constants import DICOMTAGS
+from six.moves import range
+
+
+#
+# This is the plugin to handle translation of DICOM SEG objects
+#
+class DICOMM3DPluginClass(DICOMM3DPluginBase):
+
+  def __init__(self):
+    super(DICOMM3DPluginClass,self).__init__()
+    self.loadType = "DICOMM3D"
+
+  def examineFiles(self,files):
+    """ Returns a list of DICOMLoadable instances
+    corresponding to ways of interpreting the
+    files parameter.
+    """
+    loadables = []
+
+    # just read the modality type; need to go to reporting logic, since DCMTK
+    #   is not wrapped ...
+
+    for cFile in files:
+
+      uid = slicer.dicomDatabase.fileValue(cFile, self.tags['instanceUID'])
+      if uid == '':
+        return []
+
+      desc = slicer.dicomDatabase.fileValue(cFile, self.tags['seriesDescription'])
+      if desc == '':
+        desc = "Unknown"
+      
+      # EncapsulatedDocument = slicer.dicomDatabase.fileValue(cFile, self.tags['EncapsulatedDocument'])
+      # EncapsulatedDocumentLength = slicer.dicomDatabase.fileValue(cFile, self.tags['EncapsulatedDocumentLength'])
+
+      isDicomM3D = (slicer.dicomDatabase.fileValue(cFile, self.tags['modality']) == 'M3D')
+
+      # print(isDicomM3D)
+      # print(desc)
+      # print(self.tags['modality'])
+      # print(cFile)
+      # print(uid)
+      if isDicomM3D:
+        loadable = DICOMLoadable()
+        loadable.files = [cFile]
+        loadable.name = desc
+        loadable.tooltip = loadable.name + ' - as a DICOM M3D object'
+        loadable.selected = True
+        loadable.confidence = 0.95
+        loadable.uid = uid
+        # loadable.EncapsulatedDocument = EncapsulatedDocument
+        # loadable.EncapsulatedDocumentLength = EncapsulatedDocumentLength 
+        self.addReferences(loadable)
+        loadables.append(loadable)
+
+        logging.debug('DICOM M3D modality found')
+
+    return loadables
+
+  def referencedSeriesName(self,loadable):
+    """Returns the default series name for the given loadable"""
+    referencedName = "Unnamed Reference"
+    if hasattr(loadable, "referencedSeriesUID"):
+      referencedName = self.defaultSeriesNodeName(loadable.referencedSeriesUID)
+    return referencedName
+
+  def GetEncapsulatedDocument(self, loadable):
+    return loadable.EncapsulatedDocument
+  
+  def GetEncapsulatedDocumentLength(self, loadable):
+    return loadable.EncapsulatedDocumentLength
+
+
+  def load(self,loadable):
+    """ Load the DICOM M3D object
+    """
+    logging.debug('DICOM M3D load()')
+    try:
+      uid = loadable.uid
+      logging.debug('in load(): uid = '+uid)
+    except AttributeError:
+      return False
+
+    self.tempDir = os.path.join(slicer.app.temporaryPath, 
+                                "QIICR", "M3D",
+                                  self.currentDateTime, loadable.uid)
+    print(self.tempDir)
+    try:
+        os.makedirs(self.tempDir)
+    except OSError:
+      pass
```

Comment:
Here you could consider using `slicer.util.tempDirectory` for brevity and to ease future maintenance.